src/logic/handlers.js

```diff
@@ -0,0 +1,321 @@
+import React, { useContext } from 'react';
+import { localStorageManager } from './local-storage-manager';
+import useStore from './store';
+import soundCorrect from "../assets/audio/correct.mp3";
+import soundIncorrect from "../assets/audio/incorrect.mp3";
+import soundGameOver from "../assets/audio/game-over.mp3";
+import soundSwitch from "../assets/audio/switch.mp3";
+import soundClose from "../assets/audio/close.mp3";
+
+const Context = React.createContext();
+export default function useHandlers() {
+  return useContext(Context);
+}
+
+export const HandlersProvider = ({ children }) => {
```

Comment:
I think this approach should be reconsidered. you can manage all your logic easier and more concise. you are combining redux-like approach from one hand and mutative logic from another. this is likely to cause bugs